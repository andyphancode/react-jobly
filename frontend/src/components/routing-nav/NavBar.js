import React, { useState, useContext } from "react";
import "./NavBar.css";
import 'bootstrap/dist/css/bootstrap.css';
import { Navbar, NavbarToggler, NavbarBrand, Collapse, Nav, NavLink, NavItem } from "reactstrap";
import UserContext from "../UserContext";


function NavBar({ logout }) {

    const { currentUser } = useContext(UserContext);
    const [isOpen, setIsOpen] = useState(false); 
    const toggle = () => { setIsOpen(!isOpen) }
    
    function loggedIn() {
      return (

        <Nav className="ms-auto" navbar>
          <NavItem className="me-4">
            <NavLink href="/companies">Companies</NavLink>
          </NavItem>
          <NavItem className="me-4">
              <NavLink href="/jobs">Jobs</NavLink>
          </NavItem>
          <NavItem className="me-4">
              <NavLink href="/profile">Profile</NavLink>
          </NavItem>
          <NavItem className="me-4">
              <NavLink href="/" onClick={logout}>Logout {currentUser.username}</NavLink>
          </NavItem>
        </Nav>          

      );      
    };

    function loggedOut() {
      return (
        <Nav className="ms-auto" navbar>
          <NavItem className="me-4">
              <NavLink href="/login">Login</NavLink>
          </NavItem>
          <NavItem className="me-4">
              <NavLink href="/signup">Sign Up</NavLink>
          </NavItem>
        </Nav>          
    );      
    }

    return (

      <Navbar className="NavBar bg-white" expand="md" >
        <NavbarBrand href="/">Jobly</NavbarBrand>
        <NavbarToggler onClick={toggle} /> 
        <Collapse isOpen={isOpen} navbar> 

            {currentUser ? loggedIn() : loggedOut()}
      
        </Collapse>
      </Navbar>
    )

  }
  
  export default NavBar;