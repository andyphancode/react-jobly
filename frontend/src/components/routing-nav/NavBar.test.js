import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import NavBar from "./NavBar";
import UserContext, { UserProvider } from "../../testProvider";


it("renders without crashing", function () {
    render(
        <MemoryRouter>
            <UserProvider>
                <NavBar />
            </UserProvider>
        </MemoryRouter>,
    );
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <NavBar />
            </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});