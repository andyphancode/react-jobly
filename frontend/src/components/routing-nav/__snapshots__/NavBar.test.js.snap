// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="NavBar bg-white navbar navbar-expand-md"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Jobly
      </a>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
      >
        <ul
          class="ms-auto navbar-nav"
        >
          <li
            class="me-4 nav-item"
          >
            <a
              class="nav-link"
              href="/companies"
            >
              Companies
            </a>
          </li>
          <li
            class="me-4 nav-item"
          >
            <a
              class="nav-link"
              href="/jobs"
            >
              Jobs
            </a>
          </li>
          <li
            class="me-4 nav-item"
          >
            <a
              class="nav-link"
              href="/profile"
            >
              Profile
            </a>
          </li>
          <li
            class="me-4 nav-item"
          >
            <a
              class="nav-link"
              href="/"
            >
              Logout test
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
